llama-index-core/llama_index/core/callbacks/token_counting.py

```diff
@@ -47,52 +87,33 @@ def get_llm_token_counts(
         response = payload.get(EventPayload.RESPONSE)
         response_str = str(response)
 
-        # try getting attached token counts first
-        try:
-            messages_tokens = 0
-            response_tokens = 0
-
-            if response is not None and response.raw is not None:
-                if isinstance(response.raw, dict):
-                    raw_dict = response.raw
-                else:
-                    raw_dict = response.raw.model_dump()
-
-                usage = raw_dict.get("usage", None)
-
-                if usage is not None:
-                    messages_tokens = usage.get("prompt_tokens", 0)
-                    response_tokens = usage.get("completion_tokens", 0)
-
-                if messages_tokens == 0 or response_tokens == 0:
-                    raise ValueError("Invalid token counts!")
-
-                return TokenCountingEvent(
-                    event_id=event_id,
-                    prompt=messages_str,
-                    prompt_token_count=messages_tokens,
-                    completion=response_str,
-                    completion_token_count=response_tokens,
-                )
+        if response and response.raw:
+            prompt_tokens, completion_tokens = get_tokens_from_raw_response(
+                response.raw
+            )
+        else:
+            prompt_tokens, completion_tokens = 0, 0
 
-        except (ValueError, KeyError):
-            # Invalid token counts, or no token counts attached
-            pass
+        if prompt_tokens == 0:
+            prompt_tokens = token_counter.get_string_tokens(messages_str)
 
-        # Should count tokens ourselves
-        messages_tokens = token_counter.estimate_tokens_in_messages(messages)
```

Comment:
do we not need to apply `estimate_tokens_in_messages` anymore?

Reply:
The diff is a little messy, but it was still there. But forget to use this special messages function. Added!